draft-protected-headers.md

```diff
@@ -28,58 +32,192 @@ author:
     email: dkg@fifthhorseman.net
 informative:
 normative:
+ RFC2119:
+ RFC2822:
  RFC3156:
+ RFC5751:
+ RFC8174:
+ RFC8551:
 --- abstract
 
-Cryptographic e-mails do not protect headers.  This document describes how several implementations protect headers.
+This document describes a common strategy to extend the cryptographic protections provided by PGP/MIME etc. to also protect message headers. In particular, how to encrypt the Subject line.
```

Comment:
I like these changes - except I don't think what we are doing is actually compatible with S/MIME. Saying so implies we are compatible with whatever the current S/MIME spec has to say about header protection, which I am pretty sure is not true.

Protected Headers might be applicable to S/MIME as well, which is what I was trying to communicate, but since there are no actual implementations out there it's largely academic. To avoid confusion on that point, we might want to just omit it from the spec entirely, aside from maybe a dedicated "Future Work" section?

Reply:
"Future work" sounds good. I actually need to read the other RFCs to have any opinion on the compatibility ;)